ludwig/models/trainer.py

```diff
@@ -1158,22 +1244,25 @@ def check_progress_on_validation(
 
         # If any early stopping condition is satisfied, then trigger early stopping behavior
         early_stop_bool = 0 < early_stop <= progress_tracker.last_improvement
+        print(f"early_stop_bool: {early_stop_bool}")
         if not early_stop_bool:
             for callback in self.callbacks:
                 if callback.should_early_stop(self, progress_tracker, self.is_coordinator()):
+                    print("callback for should_early_stop was true")
                     early_stop_bool = True
                     break
 
         # ========== Early Stop logic ==========
         should_early_stop = torch.as_tensor([early_stop_bool], dtype=torch.int)
+        print(f"should_early_stop: {should_early_stop}")
```

Comment:
Looks like we're missing the `should_early_stop` callback I added. Will need to pull that in.